starlite/kwargs.py

```diff
@@ -499,26 +508,94 @@ def to_kwargs(self, connection: Union["WebSocket", "Request"]) -> Dict[str, Any]
+        """Resolve all dependencies into the kwargs, recursively.
 
         Args:
-            dependency: An instance of [Dependency][starlite.kwargs.Dependency]
             connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
-            **kwargs: Any kwargs to pass recursively.
+            kwargs: Kwargs to pass to dependencies
+        """
+        for batch in self.dependency_batches:
+            await self._resolve_dependencies_concurrently(batch, connection, kwargs)
 
-        Returns:
-            The resolved dependency value
+    async def _resolve_dependencies_concurrently(
+        self,
+        dependencies: Set["Dependency"],
+        connection: Union["WebSocket", "Request"],
+        kwargs: Dict[str, Any],
+    ) -> None:
+        """Resolve the dependencies concurrently. The passed in kwargs will be modified with the results.
+
+        Args:
+            dependencies: An iterable for instances of [Dependency][starlite.kwargs.Dependency].
+            connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
+            kwargs: Kwargs to pass to dependencies
+        """
+        if not dependencies:
+            pass
+        elif len(dependencies) == 1:
+            await self._resolve_dependency(next(iter(dependencies)), connection, kwargs)
+        else:
+            async with create_task_group() as task_group:
+                for dependency in dependencies:
+                    task_group.start_soon(self._resolve_dependency, dependency, connection, kwargs)
+
+    @staticmethod
+    async def _resolve_dependency(
+        dependency: "Dependency", connection: Union["WebSocket", "Request"], kwargs: Dict[str, Any]
+    ) -> None:
+        """Resolve a given instance of [Dependency][starlite.kwargs.Dependency]. All required sub dependencies must
+        already be resolved into the kwargs.
+
+        Args:
+            dependency: An instance of [Dependency][starlite.kwargs.Dependency]
+            connection: An instance of [Request][starlite.connection.Request] or [WebSocket][starlite.connection.WebSocket].
+            kwargs: Any kwargs to pass recursively.
         """
         signature_model = get_signature_model(dependency.provide)
-        for sub_dependency in dependency.dependencies:
-            kwargs[sub_dependency.key] = await self.resolve_dependency(
-                dependency=sub_dependency, connection=connection, **kwargs
-            )
         dependency_kwargs = signature_model.parse_values_from_connection_kwargs(connection=connection, **kwargs)
-        return await dependency.provide(**dependency_kwargs)
+        kwargs[dependency.key] = await dependency.provide(**dependency_kwargs)
+
+    @classmethod
+    def _get_dependency_batches(cls, expected_dependencies: Set[Dependency]) -> List[Set[Dependency]]:
```

Comment:
same as previous, these two class methods should be moved out of the class. 

Regarding names:

instead of `_get_dependency_batches` --> `batch_dependencies`, `split_dependencies_into_batches`, `create_dependency_batches etc.` 

`_get_dependency_batches_helper ` --> this name is meaningless. Try ` map_dependencies_recurisvely` or something like this.